integrations/gitlab/gitlab_integration/events/event_handler.py

```diff
@@ -1,37 +1,77 @@
 import asyncio
+from abc import abstractmethod, ABC
+from asyncio import Queue
 from collections import defaultdict
+from copy import deepcopy
+
 from loguru import logger
 from typing import Awaitable, Callable, Any, Type
 
 from gitlab_integration.events.hooks.base import HookHandler
 from gitlab_integration.gitlab_service import GitlabService
+from port_ocean.context.event import event as current_event, event_context, EventContext
 
 Observer = Callable[[str, dict[str, Any]], Awaitable[Any]]
 
 
-class EventHandler:
+class BaseEventHandler(ABC):
     def __init__(self) -> None:
+        self.webhook_tasks_queue: Queue[tuple[EventContext, str, dict[str, Any]]] = (
+            Queue()
+        )
+
+    async def start_event_processor(self) -> None:
+        logger.info(f"Started {self.__class__.__name__} worker")
+        while True:
```

Comment:
check the application is not stuck on shutdown

Reply:
it stuck but also on main branch